src/Passes/cmake/Testing.cmake

```diff
@@ -0,0 +1,46 @@
+if (MICROSOFT_ENABLE_TESTS)
+  include(CTest)
+  enable_testing()
+endif (MICROSOFT_ENABLE_TESTS)
+
+
+function (_internal_add_test name source library)
+  if (MICROSOFT_ENABLE_TESTS)
+    file(GLOB_RECURSE srcs RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} ${source}/*.cpp)
+    file(GLOB_RECURSE hdrs RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} ${source}/*.hpp)
+    file(GLOB_RECURSE ipps RELATIVE ${CMAKE_CURRENT_SOURCE_DIR} ${source}/*.ipp)
+
+    add_executable(${name} ${ipps} ${hdrs} ${srcs})
+#    target_link_libraries(${name} PRIVATE ${library} gmock gmock_main)  
+    target_include_directories(${name} 
+                                 PRIVATE ${MICROSOFT_ROOT_VENDOR_DIR}/googletest/googlemock/include)
+
+    add_test(${name}
+             ${name}
+             --gtest_shuffle
+             --gtest_random_seed=1337)  
+  endif (MICROSOFT_ENABLE_TESTS)
+endfunction()
+
+function (microsoft_add_library_tests
+          library)
+  if (MICROSOFT_ENABLE_TESTS)
+    list(REMOVE_AT ARGV 0)
```

Comment:
Here again, I'm don't see why removing the first item from `ARGV` is needed.

Reply:
> Regarding the test dependency on an external submodule, would it be possible to just use Catch2 like the qsharp-runtime repo? See https://github.com/microsoft/qsharp-runtime/tree/main/src/Qir/Common/Externals#catch2 for where we use the library, and https://github.com/microsoft/qsharp-runtime/blob/main/src/Qir/Tests/QIR-dynamic/qir-driver.cpp as an example of how we use it.

What I agreed with @bettinaheim is that we follow the standard used by LLVM and try to align with them as closely as possible. That said, I am happy for us to use Catch2 instead of gmock if the consensus is that this is a better choice.